src/useChat.tsx

```diff
@@ -151,27 +162,43 @@ export const useChat = (initValues?: ChatMessage[], initModel?: string) => {
 
   const onSelectConversation = useCallback(
     async (id: string) => {
-      const selectedConversation = conversationHistories[id];
-      if (selectedConversation) {
-        const messages = await getConversationMessages(id);
-
-        setActiveChat((prev) => ({
-          ...prev,
-          id: selectedConversation.id,
-          model: selectedConversation.model,
-          isConversationStarted: true,
-          messageHistoryStore: new MessageHistoryStore(
-            messages ? messages : [],
-          ),
-        }));
+      // already selected
+      if (id === activeChat.id) return;
+
+      if (activeChats[id]) {
+        setActiveChat(activeChats[id]);
+      } else {
+        const selectedConversation = conversationHistories[id];
+        if (selectedConversation) {
+          const messages = await getConversationMessages(id);
+          const newActiveChat: ActiveChat = {
+            id: selectedConversation.id,
+            model: selectedConversation.model,
+            isRequesting: false,
+            isConversationStarted:
+              Array.isArray(messages) &&
+              messages.find((msg) => msg.role === 'assistant') !== undefined,
+            messageHistoryStore: new MessageHistoryStore(
+              messages ? messages : [],
+            ),
```

Comment:
Could this be shortened to 
```suggestion
            messageHistoryStore: new MessageHistoryStore(
              messages ?? [],
            ),
```